extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/JacksonMessageBodyReader.java

```diff
@@ -56,4 +56,15 @@ private Object doReadFrom(Class<Object> type, Type genericType, InputStream enti
         return reader.forType(reader.getTypeFactory().constructType(genericType != null ? genericType : type))
                 .readValue(entityStream);
     }
+
+    private boolean isReadable(MediaType mediaType, Class<?> type, Type genericType) {
+        if (mediaType == null) {
+            return false;
+        }
+        if (String.class.equals(type) && (String.class.equals(genericType) || (genericType == null))) { // don't attempt to read plain strings
+            return false;
+        }
+        String subtype = mediaType.getSubtype();
+        return "json".equalsIgnoreCase(subtype) || subtype.endsWith("+json") || "javascript".equals(subtype);
```

Comment:
Really? `javascript`?

Reply:
Well, I copied this from some of JAX-RS specific snippet in the Jackson codebase. I thought it was odd, but said 🤷🏼 :)

Reply:
It _is_ odd. https://stackoverflow.com/a/477819/56994 says `application/javascript` is for JSONP. This doesn't even check the main type. It feels wrong.

Reply:
OK, I'll remove it